### PR TITLE
Save memory in on aggs in async search

### PR DESCRIPTION
**Pull request by nik9000**
This replaces a reference to the result of partially reducing
aggregations that async search keeps with a reference to the serialized
form of the result of the partial reduction which we need to keep
anyway.

### CODE REVIEW

**Comment by nik9000** [comment]
I've marked this a blocker for 7.8.0 because we'd really like to avoid releasing with the aggs referenced twice in async search. We have plenty of time to get it or something like it merged.

**Comment by nik9000** [comment]
Follow up to https://github.com/elastic/elasticsearch/pull/54758#pullrequestreview-390018326

**Comment by nik9000** [comment]
Test failures are real. Looks like I should have run more of the tests before opening!

**Comment by nik9000** [comment]
I could have made aggs from remote clusters delayed but that'd have effected the transport client and, maybe, other things. I'll investigate it, but I don't want to "sneak" that in.

File: server/src/main/java/org/elasticsearch/search/internal/InternalSearchResponse.java [hunk 1]
```diff
@@ -43,32 +44,66 @@ public static InternalSearchResponse empty(boolean withTotalHits) {
...
         out.writeVInt(numReducePhases);
     }
+
+    @Override
+    public InternalAggregations aggregations() {
+        if (hasAggregations) {
+            if (aggregations == null) {
+                throw new IllegalStateException("aggregations already consumed");
+            }
+            return aggregations.get();
```
Comment (jimczi) [inline server/src/main/java/org/elasticsearch/search/internal/InternalSearchResponse.java:95]: I am worried about the hidden cost of calling this on `serialized` aggs. Can we at least ensure that we don't get aggregations twice like [here](https://github.com/Kobar-Demo/Elasticsearch_Clone/blob/master/x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/MutableSearchResponse.java#L132) ?

**Review by jimczi — CHANGES_REQUESTED** [review]
The change makes sense to me. I left one comment regarding the usage of `InternalSearchResponse#aggregations` that should be addressed.

**Comment by nik9000** [comment]
Replaced by #55683.

**Pull request closed without merging**